pkg/interceptors/github/github.go

```diff
@@ -96,7 +129,306 @@ func (w *Interceptor) Process(ctx context.Context, r *triggersv1.InterceptorRequ
 		}
 	}
 
+	// For event types pull_request, issue_comment check github owners approval is required
+	// User can specify both event type or any one of them
+	if p.GithubOwners.Enabled {
+		ownerCheckAllowed := false
+		for _, allowedEvent := range ownersEventTypes {
+			if actualEvent == allowedEvent {
+				ownerCheckAllowed = true
+				break
+			}
+		}
+		if ownerCheckAllowed {
+			ghToken, err := w.getPersonalAccessTokenSecret(ctx, r, p)
+			if err != nil {
+				return interceptors.Failf(codes.FailedPrecondition, "error getting github token: %v", err)
+			}
+			if ghToken == "" && (p.GithubOwners.EnableOrgMemberCheck || p.GithubOwners.EnableRepoMemberCheck) {
+				return interceptors.Fail(codes.FailedPrecondition, "enableOrgMemberCheck or enableRepoMemberCheck is enabled but no personalAccessToken was supplied")
+			}
+
+			enterpriseBaseURL := headers.Get("X-Github-Enterprise-Host")
```

Comment:
Does it mean this will only work for GH enterprise and not regular hosted GitHub? (Could we just default to github.com of `X-Github-Enterprise-Host` is not specified?

@kcloutie I think this applies to the add changed files feature too?

Reply:
Yes, the same method is being used in the add changed files feature. This code works with both enterprise and regular hosted github. The [X-Github-Enterprise-Host](https://docs.github.com/enterprise-server@3.5/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#delivery-headers) header only exists when the webhook comes from a github enterprise server. So if the header does not exist, the `enterpriseBaseURL` is an empty string. We assume if the string is empty, it is regular hosted GitHub and if it has a value, it is enterprise. See [here](https://github.com/nagasree9/triggers/blob/d36e00e85cea2533e4d9a143c21a769ff477441c/pkg/interceptors/github/github.go#L346-L360). The github library we are using only asks for the guthub server url for enterprise.

I think a comment around this line explaining what is happening would help a great deal

Reply:
Ah my bad, I missed the part where if the enterpriseBasedURL is empty, we just initialize a regular client.
I do agree that a comment would help!